src/main/java/com/launchdarkly/client/LDClient.java

```diff
@@ -174,33 +184,42 @@ private void sendFlagRequestEvent(String featureKey, LDUser user, JsonElement va
   }
 
   /**
-   * Returns a map from feature flag keys to Boolean feature flag values for a given user. The map will contain {@code null}
-   * entries for any flags that are off or for any feature flags with non-boolean variations. If the client is offline or
-   * has not been initialized, a {@code null} map will be returned.
+   * Returns a map from feature flag keys to {@code JsonElement} feature flag values for a given user.
+   * If the result of a flag's evaluation would have returned the default variation, it will have a null entry
+   * in the map. If the client is offline, has not been initialized, or a null user or user with null/empty user key a {@code null} map will be returned.
    * This method will not send analytics events back to LaunchDarkly.
    * <p>
    * The most common use case for this method is to bootstrap a set of client-side feature flags from a back-end service.
    *
    * @param user the end user requesting the feature flags
-   * @return a map from feature flag keys to JsonElement values for the specified user
+   * @return a map from feature flag keys to {@code JsonElement} for the specified user
    */
-  public Map<String, Boolean> allFlags(LDUser user) {
+  public Map<String, JsonElement> allFlags(LDUser user) {
```

Comment:
I think the changes should be in the Github release- I'll start a draft for 2.0.0 now.